src/architecture/utilities/tests/CMakeLists.txt

```diff
@@ -1,3 +1,9 @@
+add_executable(test_discretize test_discretize.cpp)
+target_link_libraries(test_discretize GTest::gtest_main)
+target_link_libraries(test_discretize ArchitectureUtilities)
+
 if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64" AND CMAKE_GENERATOR STREQUAL "Xcode")
     set(CMAKE_GTEST_DISCOVER_TESTS_DISCOVERY_MODE PRE_TEST)
 endif()
+
+gtest_discover_tests(test_discretize)
```

Comment:
Is there a way to automate test discovery? This seems like a potential cause for future mistakes: someone forgetting to add their test to the CMakeLists.

If this is not possible, maybe we have a function that wraps these lines into one?
`add_gtest(test_discretize)` would add the executable, link the libraries and call the `gtest_discover_tests`.

Reply:
I wasn't able to find an integration for test discovery across the whole project. That's not to say that it isn't possible. Agreed on the adding of a Cmake function.